src/vs/workbench/contrib/mergeEditor/browser/commands/commands.ts

```diff
@@ -361,20 +362,33 @@ export class CompareInput2WithBaseCommand extends MergeEditorAction {
 	}
 
 	override runWithViewModel(viewModel: MergeEditorViewModel, accessor: ServicesAccessor): void {
-		const commandService = accessor.get(ICommandService);
-		mergeEditorCompare(viewModel, commandService, 2);
+		const editorService = accessor.get(IEditorService);
+		mergeEditorCompare(viewModel, editorService, 2);
 	}
 }
 
-function mergeEditorCompare(viewModel: MergeEditorViewModel, commandService: ICommandService, inputNumber: 1 | 2) {
+async function mergeEditorCompare(viewModel: MergeEditorViewModel, editorService: IEditorService, inputNumber: 1 | 2) {
 	const model = viewModel.model;
-	const base = model.base.uri;
-	const input = inputNumber === 1 ? model.input1.textModel.uri : model.input2.textModel.uri;
-	openDiffEditor(commandService, base, input);
-}
-
-function openDiffEditor(commandService: ICommandService, left: URI, right: URI, label?: string) {
-	commandService.executeCommand(API_OPEN_DIFF_EDITOR_COMMAND_ID, left, right, label);
+	const base = model.base;
+	const input = inputNumber === 1 ? viewModel.inputCodeEditorView1.editor : viewModel.inputCodeEditorView2.editor;
+
+	const lineNumber = input.getPosition()!.lineNumber;
+	const editor = await editorService.openEditor({
+		original: { resource: base.uri },
+		modified: { resource: input.getModel()!.uri },
+	});
+	const e = editor?.getControl();
+	if (!e) {
+		return;
+	}
+	const d = ((e as any).onDidUpdateDiff as Event<unknown>)(() => {
```

Comment:
This shouldn't be so complicated. I am certain that the `openEditor` call or the input itself supports the passing and revealing of a position. @lramos15 for details

Reply:
@hediet And I discussed offline and have settled on using the selection options for this